agent/api/task.go

```diff
@@ -464,6 +464,25 @@ func (task *Task) dockerConfig(container *Container) (*docker.Config, *DockerCli
 	return config, nil
 }
 
+// SetConfigHostConfigCommonFileds sets the same fields in both Config and HostConfig for backward compatibility
+func (task *Task) SetConfigHostConfigCommonFileds(container *Container, config *docker.Config, hc *docker.HostConfig) {
+	// Convert MB to B
+	dockerMem := int64(container.Memory * 1024 * 1024)
```

Comment:
I think we have a const (`bytesPerMegabyte`) for this multiplier in `task_unix.go`. 
Could be moved and reused if necessary.